spec/nxt_state_machine_spec.rb

```diff
@@ -0,0 +1,215 @@
+RSpec.describe NxtStateMachine do
+
+  let(:article) { Article.new(type: 'manual_approval') }
+  let(:headline) { 'Article about state machines' }
+
+  subject do
+    ArticleWorkflow.new(article, test: 'options')
```

Comment:
Not really clear where this class comes from: the spec is stated to describe `NxtStateMachine` but the subject is `ArticleWorkflow`. It would help to define `ArticleWorkflow` here explicitly.